core/src/main/kotlin/com/github/spigotbasics/core/command/parsed/ParsedCommandBuilder.kt

```diff
@@ -0,0 +1,96 @@
+package com.github.spigotbasics.core.command
+
+import com.github.spigotbasics.common.Either
+import com.github.spigotbasics.core.command.parsed.ArgumentPath
+import com.github.spigotbasics.core.command.parsed.Command
+import com.github.spigotbasics.core.command.parsed.CommandContext
+import com.github.spigotbasics.core.command.parsed.CommandExecutor
+import com.github.spigotbasics.core.messages.Message
+import com.github.spigotbasics.core.module.BasicsModule
+import org.bukkit.permissions.Permission
+
+class ParsedCommandBuilder<T : CommandContext>(
+    private val module: BasicsModule,
+    private val name: String,
+    private val permission: Permission,
+) {
+    private var permissionMessage: Message = module.plugin.messages.noPermission
+    private var description: String? = null
+    private var usage: String = ""
+    private var aliases: List<String> = emptyList()
+    private var executor: BasicsCommandExecutor? = null
+    private var tabCompleter: BasicsTabCompleter? = null
+    private var parsedExecutor: CommandExecutor<T>? = null
+    private var argumentPaths: List<ArgumentPath<T>>? = null
+
+    fun description(description: String) = apply { this.description = description }
+
+    fun usage(usage: String) =
+        apply {
+            if (usage.startsWith("/")) error("Usage should not start with /<command> - only pass the arguments.")
+            this.usage = usage
+        }
+
+    fun paths(argumentPaths: List<ArgumentPath<T>>) = apply { this.argumentPaths = argumentPaths }
```

Comment:
Not to be picky, but surely it would look nicer with a `varargs ArgumentPath<T> = apply { this.argumentPaths = argumentPaths.asList()`

Reply:
Yes that would probably be better!